src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -215,7 +216,8 @@ void CascadiaSettings::_ValidateProfilesHaveGuid()
 {
     for (auto& profile : _profiles)
     {
-        profile.GenerateGuidIfNecessary();
+        auto profileImpl = winrt::get_self<implementation::Profile>(profile);
+        profileImpl->GenerateGuidIfNecessary();
     }
 }

```

Comment:
in the future, we should probably just make Profile itself handle this directly _during `Guid()`_. @zadjii-msft  thoughts?

Reply:
Hmm...but then `Guid()` isn't a const function. Could we move this to the constructor instead? Then completely get rid of `HasGuid()` because we'll always have a guid?

Reply:
wait don't do that, I'm pretty sure there's a reason that a profile doesn't _initially_ have a GUID, and we always assign it later (though I can't remember off the top of my head). 

I like the idea of `Guid()` generating the GUID when it's needed, but let's not muck with it for now. I think this should be #resolution-maybe-we'll-fix-later

Reply:
Opened a new issue to track this: https://github.com/microsoft/terminal/issues/7421